src/parser/postgresparser.cpp

```diff
@@ -1498,16 +1501,25 @@ std::unique_ptr<SQLStatement> PostgresParser::CreateViewTransform(ParseResult *p
 PostgresParser::ColumnDefTransResult PostgresParser::ColumnDefTransform(ParseResult *parse_result, ColumnDef *root) {
   auto type_name = root->type_name_;
 
-  // handle varlen
-  int32_t varlen = -1;
+  auto datatype_name = reinterpret_cast<value *>(type_name->names_->tail->data.ptr_value)->val_.str_;
+  auto datatype = ColumnDefinition::StrToDataType(datatype_name);
+
+  // handle varlen and fixed decimal precision
+  size_t varlen = 0;
+
   if (type_name->typmods_ != nullptr) {
     auto node = reinterpret_cast<Node *>(type_name->typmods_->head->data.ptr_value);
     switch (node->type) {
       case T_A_Const: {
         auto node_type = reinterpret_cast<A_Const *>(node)->val_.type_;
         switch (node_type) {
           case T_Integer: {
-            varlen = static_cast<int32_t>(reinterpret_cast<A_Const *>(node)->val_.val_.ival_);
+            if (datatype == ColumnDefinition::DataType::DECIMAL) {
```

Comment:
Do you only want to do this for DECIMAL? What if we just always do this?

Reply:
Differently phrased, do you know what this is for e.g. INTEGER?

Reply:
it is not present